perf/graaflib/bron_kerbosch_benchmark.cpp

```diff
@@ -0,0 +1,245 @@
+      graph.add_edge(vertices[i], vertices[i + 1], 1);
+    }
+  }
+
+  graaf::algorithm::bron_kerbosch(graph);
+  state.SetComplexityN(state.range(0));
+}
+
+static void bron_kerbosh_dense_graph(benchmark::State& state) {
+  const auto number_of_edges{static_cast<size_t>(state.range(0))};
+
+  graaf::undirected_graph<int, int> graph{};
+
+  // We create enough vertices to construct the requested number of edges
+  const auto number_of_vertices{number_of_edges + 1};
+  const auto vertices{create_vertices(graph, number_of_vertices)};
+
+  for (auto _ : state) {
+    for (size_t i{0}; i < number_of_edges; ++i) {
+      for (size_t j{i + 1}; j < number_of_edges; ++j) {
+        if (i != j && !graph.has_edge(vertices[i], vertices[j]))
+          graph.add_edge(vertices[i], vertices[j], 1);
+      }
+    }
+  }
+
+  graaf::algorithm::bron_kerbosch(graph);
+  state.SetComplexityN(state.range(0));
+}
+
+static void bron_kerbosh_graph_triangle_cliques(benchmark::State& state) {
+  const auto number_of_edges{static_cast<size_t>(state.range(0))};
+
+  graaf::undirected_graph<int, int> graph{};
+
+  // We create enough vertices to construct the requested number of edges
+  const auto number_of_vertices{number_of_edges + 1};
+  const auto vertices{create_vertices(graph, number_of_vertices)};
+
+  size_t clique_size = 4;
+
+  for (auto _ : state) {
+    for (size_t i{0}; i < number_of_edges; i += clique_size) {
+      std::vector<graaf::edge_id_t> clique{};
+      add_clique(graph, vertices, i, clique_size);
+    }
+  }
+
+  graaf::algorithm::bron_kerbosch(graph);
+  state.SetComplexityN(state.range(0));
+}
+
+static void bron_kerbosh_graph_connected_triangle_cliques(
+    benchmark::State& state) {
+  const auto number_of_edges{static_cast<size_t>(state.range(0))};
+
+  graaf::undirected_graph<int, int> graph{};
+
+  // We create enough vertices to construct the requested number of edges
+  const auto number_of_vertices{number_of_edges + 1};
+  const auto vertices{create_vertices(graph, number_of_vertices)};
+
+  size_t clique_size = 4;
+
+  for (auto _ : state) {
+    // Connecting all cliques
+    for (size_t i{0}; i + clique_size < number_of_edges; i += clique_size) {
+      graph.add_edge(vertices[i], vertices[i + clique_size], 1);
+    }
+
+    for (size_t i{0}; i < number_of_edges; i += clique_size) {
+      std::vector<graaf::edge_id_t> clique{};
+      add_clique(graph, vertices, i, clique_size);
+    }
+  }
+
+  graaf::algorithm::bron_kerbosch(graph);
+  state.SetComplexityN(state.range(0));
+}
+
+static void bron_kerbosh_graph_large_cliques(benchmark::State& state) {
+  const auto number_of_edges{static_cast<size_t>(state.range(0))};
+
+  graaf::undirected_graph<int, int> graph{};
+
+  // We create enough vertices to construct the requested number of edges
+  const auto number_of_vertices{number_of_edges + 1};
+  const auto vertices{create_vertices(graph, number_of_vertices)};
+
+  size_t clique_size = 25;
```

Comment:
We could also do this in a follow-up, but looking at these parameters it looks like we can pull out the `clique_size` as a second parameter. Then we might be able to consolidate a few of these benchmarks. I.e.

```c++
BENCHMARK(bron_kerbosh_connected_cliques)
    ->Ranges({{100, 10000}, {3, 100}})  // total number of edges, clique size
    ->Unit(benchmark::kMillisecond)
    ->Complexity();
```

Reply:
If you want I would like to add this in currecnt PR!